sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/tracing/TraceTokenTest.java

```diff
@@ -0,0 +1,179 @@
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junitpioneer.jupiter.SetSystemProperty;
+import org.sdase.commons.spring.boot.web.testing.auth.DisableSdaAuthInitializer;
+import org.sdase.commons.spring.boot.web.tracing.test.TracingTestApp;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.system.CapturedOutput;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+
+@SetSystemProperty(key = "enable.json.logging", value = "true")
+@SpringBootTest(
+    classes = TracingTestApp.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {
+      "otherServiceClient.baseUrl=http://localhost:${wiremock.server.port}/api",
+    })
+@AutoConfigureWireMock(port = 0)
+@ContextConfiguration(initializers = DisableSdaAuthInitializer.class)
+@ExtendWith(OutputCaptureExtension.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
+class TraceTokenTest {
+  @LocalServerPort private int port;
+  @Autowired private TestRestTemplate client;
+  @Autowired ObjectMapper objectMapper;
+  static final TypeReference<Map<String, Object>> MAP_REF = new TypeReference<>() {};
+
+  @BeforeEach
+  void setUp() {
+    WireMock.reset();
+  }
+
+  @Test
+  void shouldPreserveTraceTokenIfPresent(CapturedOutput output) {
+    // given
+    var headers = new HttpHeaders();
+    headers.set("Trace-Token", "pre-defined-trace-token");
+
+    // when
+    ResponseEntity<String> responseEntity = executeRequestWithHeader("/api/static", port, headers);
+
+    // then
+    assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+    // response should have same trace-token header
+    assertThat(responseEntity.getHeaders().getFirst("Trace-Token"))
+        .isEqualTo("pre-defined-trace-token");
+
+    // mdc should log same trace-token
+    var expectedLogLine = findLogLine(output);
+    assertThat(expectedLogLine).isPresent();
+    var actualLogLine = expectedLogLine.get();
+    var mdc = (Map<String, String>) toObject(actualLogLine).get("mdc");
+    assertThat(mdc)
+        .extracting("requestHeaderTraceToken", "responseHeaderTraceToken")
+        .containsExactly("pre-defined-trace-token", "pre-defined-trace-token");
```

Comment:
We could try to wrap this and [the other failing test on windows](https://github.com/SDA-SE/sda-spring-boot-commons/pull/186/files#diff-980f0a9f7c2061dede07ad67c0b5fbda62aaea8f7a55bf0d556f54fb361988d4R107-R113) in a `await().unitlAsserted(() -> {...}`.